pallets/did/src/did_details.rs

```diff
@@ -542,18 +542,14 @@ impl<T: Config> TryFrom<(DidCreationOperation<T>, DidVerificationKey)> for DidDe
 }
 
 // Generates a new DID entry starting from the current one stored in the
-// storage and by applying the changes in the [DidUpdateOperation].
+// storage and by applying the changes in the [DidUpdateDetails].
 //
 // The operation fails with a [DidError] if the update operation instructs to
 // delete a verification key that is not associated with the DID.
-//
-// Please note that this method does not perform any checks regarding
-// the validity of the [DidUpdateOperation] signature nor whether the nonce
-// provided is valid.
-impl<T: Config> TryFrom<(DidDetails<T>, DidUpdateOperation<T>)> for DidDetails<T> {
+impl<T: Config> TryFrom<(DidDetails<T>, DidUpdateDetails<T>)> for DidDetails<T> {
```

Comment:
Can we move that into a function in `impl DidDetails` that is named something like `update`? This looks like an abuse of the `TryFrom` trait

Reply:
I would also accept it if you say that this is out of scope for this PR...

Reply:
https://github.com/KILTprotocol/mashnet-node/pull/231/commits/d8bddbf9ca78a7d0f3084ee0eac1f781755e782e